### ISSUE TITLE
Plugin uninstallation error

### ISSUE THREAD

**Issue by sanak** [gtt-project/redmine_admissions#5]
**Problem**
Plugin uninstallation was impossible due to the following error.
```
% bundle exec rake redmine:plugins:migrate VERSION=0 NAME=redmine_admissions  
== 20180912100600 AddForeignKeysToAdmissionAssignedRoles: reverting ===========
-- remove_foreign_key(:admission_assigned_roles, {:column=>:role_id})
   -> 0.0128s
-- remove_foreign_key(:admission_assigned_roles, {:column=>:project_id})
   -> 0.0032s
== 20180912100600 AddForeignKeysToAdmissionAssignedRoles: reverted (0.0230s) ==

== 20180912100500 DropAdmissionAssignedRoleId: reverting ======================
-- add_reference(:projects, :admission_assigned_role)
   -> 0.0137s
== 20180912100500 DropAdmissionAssignedRoleId: reverted (0.0138s) =============

== 20180818082023 MigrateAdmissionAssignedRoles: reverting ====================
== 20180818082023 MigrateAdmissionAssignedRoles: reverted (0.0000s) ===========

== 20180818081502 CreateAdmissionAssignedRoles: reverting =====================
-- drop_table(:admission_assigned_roles, {:id=>false})
   -> 0.0010s
== 20180818081502 CreateAdmissionAssignedRoles: reverted (0.0011s) ============

== 20170915125235 AddAdmissionAssignedRoleToProjects: reverting ===============
-- remove_foreign_key(:projects, {:column=>:admission_assigned_role_id})
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Table 'projects' has no foreign key for {:column=>:admission_assigned_role_id}
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/schema_statements.rb:1329:in `foreign_key_for!'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/schema_statements.rb:991:in `remove_foreign_key'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:871:in `block in method_missing'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:840:in `block in say_with_time'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:840:in `say_with_time'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:860:in `method_missing'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:687:in `block in revert'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:686:in `each'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:686:in `revert'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:812:in `exec_migration'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:798:in `block (2 levels) in migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:797:in `block in migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/connection_pool.rb:416:in `with_connection'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:796:in `migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:977:in `migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1292:in `block in execute_migration_in_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1343:in `block in ddl_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/transactions.rb:212:in `transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1343:in `ddl_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1262:in `each'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1210:in `block in migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1210:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:514:in `down'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1015:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:536:in `migrate_plugin'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:477:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:488:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/tasks/redmine.rake:135:in `block (3 levels) in <top (required)>'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/sanak/.rbenv/versions/2.6.8/bin/bundle:23:in `load'
/Users/sanak/.rbenv/versions/2.6.8/bin/bundle:23:in `<main>'

Caused by:
ArgumentError: Table 'projects' has no foreign key for {:column=>:admission_assigned_role_id}
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/schema_statements.rb:1329:in `foreign_key_for!'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/schema_statements.rb:991:in `remove_foreign_key'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:871:in `block in method_missing'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:840:in `block in say_with_time'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:840:in `say_with_time'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:860:in `method_missing'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:687:in `block in revert'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:686:in `each'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:686:in `revert'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:812:in `exec_migration'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:798:in `block (2 levels) in migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:797:in `block in migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/connection_pool.rb:416:in `with_connection'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:796:in `migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:977:in `migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1292:in `block in execute_migration_in_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1343:in `block in ddl_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/transactions.rb:212:in `transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1343:in `ddl_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1262:in `each'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1210:in `block in migrate'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1210:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:514:in `down'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/activerecord-5.2.5/lib/active_record/migration.rb:1015:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:536:in `migrate_plugin'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:477:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/redmine/plugin.rb:488:in `migrate'
/Users/sanak/Build/rb/redmine40/lib/tasks/redmine.rake:135:in `block (3 levels) in <top (required)>'
/Users/sanak/Build/rb/redmine40/vendor/bundle/ruby/2.6.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/sanak/.rbenv/versions/2.6.8/bin/bundle:23:in `load'
/Users/sanak/.rbenv/versions/2.6.8/bin/bundle:23:in `<main>'
Tasks: TOP => redmine:plugins:migrate
(See full trace by running task with --trace)
```

**To Reproduce**
Execute the following command.
```bash
$ bundle exec rake redmine:plugins:migrate VERSION=0 NAME=redmine_admissions 
```

**Expectation**
Plugin's uninstallation should be finished without error.

**Comment by sanak** [comment]
Reopen from the following issue.
https://github.com/gtt-project/redmine_admissions/pull/6#issuecomment-1137884398

**Comment by sanak** [comment]
Fixed this again by #7. Close this.
